compserv/hknweb
# Refactor and Semester date ranges for challenges and bitbytes + Loop Merge Unsupported

Moved some big logic for challenges and bit bytes to utils.py

Added the ability to select only a subset of challenges and bitbytes based on date ranges that are determined by the Candidate Semester of the user in question (set in their settings)

Testing Instructions:
1. Recommended to load a sqlite3 backup to your local hknweb, as to confirm challenges requires sending email (currently unsure how to bypass, but could go line by line commenting out the CSec and Requester email sends)
2. Check between different candidate semesters and make sure that 1) there are dates in the respective Requirement models to denote start and end dates to accept Bit Bytes or Challenges, 2) there are entries for the specific requirements
